### PR TITLE
v2: Simplify setup, remove "transactionFactory".

### PR DESCRIPTION
**Pull request by andreibancioiu**
- The Web Wallet provider does not receive a `TransactionFactory` in its constructor anymore.
 - The transaction objects returned by the Web Wallet provider are plain JavaScript objects, not erdjs `Transaction` objects.
 - Remove `sendTransaction()`
 - Adjust `signTransaction()` to rely on `signTransactions()`

### CODE REVIEW

File: src/walletProvider.ts [hunk 1]
```diff
@@ -1,110 +1,74 @@
...
+        let callbackUrl = options?.callbackUrl || window.location.href;
+        const redirectUrl = `${this.baseWalletUrl()}${WALLET_PROVIDER_DISCONNECT_URL}?callbackUrl=${callbackUrl}`;
 
-        const redirect = `${this.baseWalletUrl()}${WALLET_PROVIDER_DISCONNECT_URL}?${callbackUrl}`;
-        // TODO: Perhaps do not delay, just like for signTransactions()?
         await new Promise((resolve) => {
             setTimeout(() => {
-              window.location.href = redirect;
-              resolve(true);
+                window.location.href = redirectUrl;
```
Comment (StanislavSava) [inline src/walletProvider.ts:54]: This looks exactly as line 37, great candidate for a separate function

File: src/walletProvider.ts [hunk 2]
```diff
@@ -1,110 +1,74 @@
...
 
     /**
     * Fetches the logout hook url and redirects the client to the wallet logout.
     */
     async logout(options?: { callbackUrl?: string }): Promise<boolean> {
-        let callbackUrl = `callbackUrl=${window.location.href}`;
-        if (options && options.callbackUrl) {
-            callbackUrl = `callbackUrl=${options.callbackUrl}`;
-        }
+        let callbackUrl = options?.callbackUrl || window.location.href;
```
Comment (StanislavSava) [inline src/walletProvider.ts:49]: This is very similar to the lines in the login function, maybe these can be turned into "getRedirectUrl(options)"

File: src/walletProvider.ts (hunk 1 above)
Comment (andreibancioiu) [reply src/walletProvider.ts:54]: Extracted to `redirectLater()`.

File: src/walletProvider.ts (hunk 2 above)
Comment (andreibancioiu) [reply src/walletProvider.ts]: Refactored URL building.

**Review by arhtudormorar — APPROVED** [review]

**Review by StanislavSava — APPROVED** [review]

**Pull request merged**